docs(tutorial) create DVC x Object storage tutorial MTA-3182 (#1732)
scaleway/docs-content: tutorials/configure-dvc-with-object-storage/index.mdx

---
meta:
  title: Configuring DVC with Object Storage

>   description: This page provides information how to configure DVC with Scaleway Object Storage.

Reviewer: ```suggestion
  description: This page provides information on how to configure DVC with Scaleway Object Storage.
```

content:
  h1: Configuring DVC with Object Storage

>   paragraph: This page provides information how to configure DVC with Scaleway Object Storage.

Reviewer: ```suggestion
  paragraph: This page provides information on how to configure DVC with Scaleway Object Storage.
```

tags: object-storage storage amazon-s3 data-version-control s3-compatible dvc-bucket machine-learning data-science
categories:
  - storage
  - object storage
dates:
  validation: 
  posted: 
---

## Data Version Control (DVC) - Overview

Git is unarguably the most popular and powerful version control system to store your code, and can handle files up to 5 GB thanks to Git LFS, but what about large data sets?

> DVC is a version control tool designed to handle large data sets. It works alongside Git and allows you to store your data in the remote storage of your choice (such as a Scaleway S3-enabled bucket) while storing only the metadata in a Git repository.

Reviewer: ```suggestion
Git is unarguably the most popular and powerful version control system to store your code and can handle files up to 5 GB thanks to Git LFS. But how to make sure it handles larger datasets? 

In these cases, you might need to turn to third-party version control tools that are designed to handle large datasets. 

One such tool is Data Version Control (DVC), which was specifically created with this use-case in mind. It works alongside Git and allows you to store your data in the remote storage of your choice (such as a Scaleway S3-enabled bucket) while storing only the metadata in a Git repository.
```
je pense qu'il faut plus directement répondre à la question que tu poses dans le premier paragraphe dans le deuxième. Du coup j'ai fait une propal où j'ai fini par changer un peu la question, mais ne te sens pas obligé de la prendre comme tel.

In this tutorial, you learn how to use [Scaleway Object Storage](https://www.scaleway.com/en/object-storage/) as a remote storage for DVC.

<Macro id="iam-requirements" />
 
<Message type="requirement">
  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
  - You have [created a bucket](/storage/object/how-to/create-a-bucket/)
  - You have [created an API key](https://www.scaleway.com/en/docs/identity-and-access-management/iam/how-to/create-api-keys/)
  - You have created a repository to store your metadata.

>   - You have done your [first request with Scaleway API](https://www.scaleway.com/en/developers/api/#quickstart:-first-request)

Reviewer: ```suggestion
  - You have made your [first request with Scaleway API](https://www.scaleway.com/en/developers/api/#quickstart:-first-request)
```

    ```bash
    dvc get https://github.com/iterative/dataset-registry \
    get-started/data.xml -o data/data.xml
    ```

2. run the following command to stage files you want to store in your bucket:
    ```bash
    dvc add data/data.xml
    ```

3. Run the following command to track the DVC metadata file and .gitignore file with git:
    ```bash
    git add data/data.xml.dvc data/.gitignore
    ```

    <Message type="note">
      The `.gitignore` file contains paths to the data files so they are not tracked by Git.
    </Message>

4. Run the following command to commit the metadata file:
    ```bash
    git commit -m "Add raw data"
    ```

5. Run the following command to push your changes to your Git repository:
   ```bash
   git push
   ```
  
## Pushing data to Scaleway Object Storage

1. Run the following command to add your bucket as a remote storage for your data:
   ```bash
   dvc remote add -d  myremote s3://my-bucket/path
   ```

2. Run the following command to set the S3 endpoint of your remote storage:
    ```bash
    dvc remote modify myremote \
    endpointurl https://s3.fr-par.scw.cloud
    ```

    <Message type="note">

> 

Reviewer: ```suggestion
```

      Edit the `endpointurl` according to the geographical region of your bucket. It can either be `fr-par` (Paris, France), `nl-ams` (Amsterdam, The Netherlands) or `pl-waw` (Warsaw, Poland).
    </Message>

3. Run the following command to push your data to your bucket:
    ```bash
    dvc push
    ```

4. Run the following command to list the content of your bucket:
    ```bash
    aws s3api list-objects-v2 --bucket my-bucket
    ```

    The content of the bucket is displayed:

    ```json
    {
      "Contents": [
          {
              "Key": "22/a1a2931c8370d3aeedd7183606fd7f",
              "LastModified": "2023-05-30T14:49:26.000Z",
              "ETag": "\"22a1a2931c8370d3aeedd7183606fd7f\"",
              "Size": 14445097,
              "StorageClass": "STANDARD"
          }
      ]
    }
    ```

    <Message type="note">
      File names are generated automatically based on the content of the `.dvc` metadata files created upon tracking the data.
    </Message>

## Going Further

> Refer to the [DVC's documentation](https://dvc.org/doc) for more information on configuration and use cases.

Reviewer: ```suggestion
Refer to the [official DVC documentation](https://dvc.org/doc) for more information on configuration and use cases.
```

> Refer to [Git's documentation](https://git-scm.com/doc) for more information and tutorials on version control. 

Reviewer: ```suggestion
Refer to [official Git documentation](https://git-scm.com/doc) for more information and tutorials on version control. 
```

tags: object-storage storage amazon-s3 data-version-control s3-compatible dvc-bucket machine-learning data-science
categories:
  - storage
  - object storage
dates:
  validation: 
  posted: 
---

## Data Version Control (DVC) - Overview

Git is unarguably the most popular and powerful version control system to store your code, and can handle files up to 5 GB thanks to Git LFS. But what about large datasets?

In that case, you might need to turn to third-party version control tools that are designed to handle large datasets. 

Data Version Control (DVC) was specifically designed with this use-case in mind. It works alongside Git and allows you to store your data in the remote storage of your choice (such as a Scaleway S3-enabled bucket) while storing only the metadata in a Git repository.

In this tutorial, you learn how to use [Scaleway Object Storage](https://www.scaleway.com/en/object-storage/) as a remote storage for DVC.

<Macro id="iam-requirements" />
 
<Message type="requirement">
  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
  - You have [created a bucket](/storage/object/how-to/create-a-bucket/)

>   - You have [created an API key](https://www.scaleway.com/en/docs/identity-and-access-management/iam/how-to/create-api-keys/)

Reviewer 2: ```suggestion
  - You have [created an API key](/identity-and-access-management/iam/how-to/create-api-keys/)
```

  - You have created a repository to store your metadata.
  - You have made your [first request with Scaleway API](https://www.scaleway.com/en/developers/api/#quickstart:-first-request)
  - You [authenticated to the API](https://www.scaleway.com/en/developers/api/#how-to-authenticate) for the first time 
  - You have installed the [AWS CLI](/storage/object/api-cli/object-storage-aws-cli/)
</Message> 

## Setting up DVC

1. Run the following command to install the DVC Python package:
    ```bash
    pip3 install dvc
    ```

2. Run the following command to install the S3 dependencies:
    ```bash
    pip3 install "dvc[s3]"
    ```

3. Run the following command in the desired repository to initialize DVC:
    ```bash
    dvc init
    ```

4. Run the following command to commit the initial DVC configuration files:
    ```bash
    git commit -m "Initialize DVC"
    ```

## Retrieving and tracking data

1. Run the following command to copy the example data file to your repository:
    ```bash

>     dvc get https://github.com/iterative/dataset-registry \

Reviewer 3: Does not work: https://github.com/iterative/dataset-registry/get-started/data.xml

Author: I just re-run the command, it works, I can't reproduce.

    </Message>

4. Run the following command to commit the metadata file:
    ```bash
    git commit -m "Add raw data"
    ```

5. Run the following command to push your changes to your Git repository:
   ```bash
   git push
   ```
  
## Pushing data to Scaleway Object Storage

1. Run the following command to add your bucket as a remote storage for your data:
   ```bash
   dvc remote add -d  myremote s3://my-bucket/path
   ```

2. Run the following command to set the S3 endpoint of your remote storage:
    ```bash
    dvc remote modify myremote \
    endpointurl https://s3.fr-par.scw.cloud
    ```

    <Message type="note">
      Edit the `endpointurl` according to the geographical region of your bucket. It can either be `fr-par` (Paris, France), `nl-ams` (Amsterdam, The Netherlands) or `pl-waw` (Warsaw, Poland).
    </Message>

3. Run the following command to push your data to your bucket:
    ```bash
    dvc push
    ```

4. Run the following command to list the content of your bucket:
    ```bash
    aws s3api list-objects-v2 --bucket my-bucket
    ```

    The content of the bucket is displayed:

    ```json
    {
      "Contents": [
          {

>               "Key": "22/a1a2931c8370d3aeedd7183606fd7f",

Reviewer 3: It looks like a hash. Are you sure it is the right data? Why is it not named data.xml ?

Author: See note below! This is the expected behavior, according to the video tutorial on their doc website: https://youtu.be/kLKBcPonMYw?t=263
